### PR TITLE
bill details view controller

### PR DESCRIPTION
**Pull request by sophiakaz**

### CODE REVIEW

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+#import "DebtorCell.h"
+
+@interface BillDetailsViewController () <UITableViewDataSource, UITableViewDelegate>
+@property (weak, nonatomic) IBOutlet UIImageView *pictureView;
+@property (weak, nonatomic) IBOutlet UILabel *memoLabel;
+@property (weak, nonatomic) IBOutlet UILabel *datelabel;
+@property (weak, nonatomic) IBOutlet UILabel *paidLabel;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray *people;
+@property (strong, nonatomic) NSDecimalNumber *sumDebt;
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
{
NSMutableArray *people;
NSDecimalNumber *sumDebt;
}
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+#import "BillDetailsViewController.h"
+#import "DebtorCell.h"
+
+@interface BillDetailsViewController () <UITableViewDataSource, UITableViewDelegate>
+@property (weak, nonatomic) IBOutlet UIImageView *pictureView;
+@property (weak, nonatomic) IBOutlet UILabel *memoLabel;
+@property (weak, nonatomic) IBOutlet UILabel *datelabel;
+@property (weak, nonatomic) IBOutlet UILabel *paidLabel;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray *people;
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    
+    //self.navigationItem.title = self.bill.memo;
+    
+    [self reloadView];
+}
+
+- (void) reloadView {
+    [self fetchPeople];
+    
+    self.sumDebt = [NSDecimalNumber zero];
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
    sumDebt = [NSDecimalNumber zero];
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    for(int i = 0; i < self.people.count - 1; i ++){
+        NSDecimalNumber *portion = [NSDecimalNumber decimalNumberWithDecimal:[self.bill.portions[i] decimalValue]];
+        self.sumDebt = [self.sumDebt decimalNumberByAdding:portion];
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
        sumDebt = [sumDebt decimalNumberByAdding:portion];
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            [imageFile getDataInBackgroundWithBlock:^(NSData * _Nullable data, NSError * _Nullable error) {
+                if (!error) {
+                    cell.pictureView.image = [UIImage imageWithData:data];
+                }
+            }];
+            cell.pictureView.layer.cornerRadius = cell.pictureView.frame.size.height /2;
+            cell.pictureView.layer.masksToBounds = YES;
+            
+            NSDecimalNumber *paid = [NSDecimalNumber decimalNumberWithDecimal:[self.bill.paid decimalValue]];
+            cell.extraMoneyLabel.text = [numberFormatter stringFromNumber:[paid decimalNumberBySubtracting:self.sumDebt]];
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
            cell.extraMoneyLabel.text = [numberFormatter stringFromNumber:[paid decimalNumberBySubtracting:sumDebt]];
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    //self.navigationItem.title = self.bill.memo;
+    
+    [self reloadView];
+}
+
+- (void) reloadView {
+    [self fetchPeople];
+    
+    self.sumDebt = [NSDecimalNumber zero];
+    for(int i = 0; i < self.people.count - 1; i ++){
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
    for(int i = 0; i < people.count - 1; i ++){
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        NSDecimalNumber *portion = [NSDecimalNumber decimalNumberWithDecimal:[self.bill.portions[i] decimalValue]];
+        self.sumDebt = [self.sumDebt decimalNumberByAdding:portion];
+    }
+    
+    [self.tableView reloadData];
+    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
+}
+
+- (void) fetchPeople {
+    self.people = [self.bill.debtors mutableCopy];
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
    people = [self.bill.debtors mutableCopy];
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    [self.people insertObject:self.bill.payer atIndex:0];
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
    [people insertObject:self.bill.payer atIndex:0];
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath{
+    
+    DebtorCell *cell = [tableView dequeueReusableCellWithIdentifier:@"DebtorCell"];
+    
+    Persona *housemate = self.people[indexPath.row];
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
    Persona *housemate = people[indexPath.row];
```

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    {
+        NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+        [dateFormatter setDateStyle:NSDateFormatterShortStyle];
+        [dateFormatter setDateFormat:@"MMMM dd, yyyy"];
+        NSString *formattedDate = [dateFormatter stringFromDate:date];
+        return formattedDate;
+    }
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.people.count;
```
Comment (jordanF0ster) [inline fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]:
```suggestion
    return people.count;
```

**Review by jordanF0ster — COMMENTED** [review]
can be bade into instance variables instead of properties

File: fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m (hunk 10 above)
Comment (sophiakaz) [reply fbu-app/PostSearch/ViewControllers/BillDetailsViewController.m]: switched them to instance variables, thanks!

**Pull request merged**